flux-refineck/src/constraint_gen.rs

```diff
@@ -54,7 +54,8 @@ pub enum Tag {
     Rem(Span),
     Goto(Option<Span>, BasicBlock),
     Overflow(Span),
-    Other,
+    Other(Span),
+    Invariant(Span),
```

Comment:
I realized we already have `Tag::Fold` so we don't this extra case. I'd make all the cases you marked with `Tag::Invariant` a `Tag::Fold`.  If you want you can change the message for `Tag::Fold` to say something about an invariant instead of fold. I see how saying fold in the error can be confusing for the user, but I would still like to refer to it as a folding error internally.

Reply:
ah yes, ok am reverting to `Fold` but changing the message for `Fold`.